cx2t/src/claw/tatsu/xcodeml/abstraction/DimensionDefinition.java

```diff
@@ -228,6 +228,27 @@ public Xnode generateIndexRange(XcodeML xcodeml, boolean withStep) {
     return range;
   }
 
+  /**
+   * Generate the correct iterationRange element with iterationLowerBound,
+   * iterationUpperBound and step from the current dimension.
+   *
+   * @param xcodeml  Current XcodeML program unit in which elements will be
+   *                 created.
+   * @param withStep IF true, step element is created.
+   * @return A new indexRange elements.
+   */
+  public Xnode generateIterationRange(XcodeML xcodeml, boolean withStep) {
```

Comment:
This method seems very similar to the one on line 219. Might be a good idea to try to combined them if possible. Or having a base method where we pass the lower and upper bound. 

```java
public Xnode generateIndexRange(XcodeML xcodeml, boolean withStep) {
```

Reply:
@clementval Yes, I agree, we can merge the two and use a flag to separate between size and iteration indices. I pushed the according change.

Reply:
Please fix the unit tests. Their compilation fails after latest change.

Reply:
Yes, sorry, had missed that. Fixed now.